java-connectors/kafka-connect-gcp-common/src/main/java/io/lenses/streamreactor/connect/gcp/common/config/GCPSettings.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2017-2024 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.gcp.common.config;
+
+import io.lenses.streamreactor.common.config.base.RetryConfig;
+import io.lenses.streamreactor.common.config.base.model.ConnectorPrefix;
+import io.lenses.streamreactor.common.config.source.ConfigSource;
+import io.lenses.streamreactor.connect.gcp.common.auth.GCPConnectionConfig;
+import io.lenses.streamreactor.connect.gcp.common.auth.HttpTimeoutConfig;
+import lombok.Getter;
+import lombok.val;
+import org.apache.kafka.common.config.ConfigDef;
+
+/**
+ * Configuration settings for connecting to Google Cloud Platform (GCP) services.
+ * This class provides methods for defining and parsing GCP-specific configuration properties.
+ */
+@Getter
+public class GCPSettings {
+
+  public static final String EMPTY_STRING = "";
+
+  private final String gcpProjectId;
+  private final String gcpQuotaProjectId;
+  private final String host;
+  private final String httpErrorRetryInterval;
+  private final String httpNbrOfRetries;
+  private final String httpSocketTimeout;
+  private final String httpConnectionTimeout;
+
+  public static final Long HTTP_ERROR_RETRY_INTERVAL_DEFAULT = 50L;
+  public static final Integer HTTP_NUMBER_OF_RETIRES_DEFAULT = 5;
+  public static final Long HTTP_SOCKET_TIMEOUT_DEFAULT = 60000L;
+  public static final Long HTTP_CONNECTION_TIMEOUT_DEFAULT = 60000L;
+
+  private final AuthModeSettings authModeSettings;
+
+  /**
+   * Constructs a new instance of {@code GCPSettings} with the specified connector prefix.
+   *
+   * @param connectorPrefix the prefix used for configuration keys
+   */
+  public GCPSettings(ConnectorPrefix connectorPrefix) {
+    gcpProjectId = connectorPrefix.prefixKey("gcp.project.id");
```

Comment:
do we resign from having separate ConfigConstants classes?

Reply:
I'm not sure if you saw my reply in [this comment](https://github.com/lensesio/stream-reactor/pull/1188#discussion_r1586301626)?

Because this code is used in more than one connector the strings need to be prefixed dynamically.

Reply:
All I'm thinking really is consistency :) See [AzureEventHubsConfigConstants](https://github.com/lensesio/stream-reactor/blob/master/java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/streamreactor/connect/azure/eventhubs/config/AzureEventHubsConfigConstants.java) : shall we follow the same logic for all of the connectors or is it case-specific? (of course such `Constant` class would have to have different prefixes into account in this scenario as multiple Connectors will use it)

Reply:
Let's leave it as it is and revisit later if we need to!